src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -344,7 +352,7 @@ def to_pex_request(
         return PexRequest(
             output_filename=f"{self.options_scope.replace('-', '_')}.pex",
             internal_only=True,
-            requirements=self.pex_requirements(extra_requirements=extra_requirements),
+            requirements=requirements,
```

Comment:
This call to self.pex_requirements() was redundant even before any of the other changes in this PR, so this is just an obvious cleanup.